src/components/structures/SpaceHierarchy.tsx

```diff
@@ -157,7 +158,8 @@ const Tile: React.FC<ITileProps> = ({
                 <Spinner w={24} h={24} />
             </AccessibleTooltipButton>
         );
-    } else if (joinedRoom) {
+    } else if (joinedRoom || room.join_rule === JoinRule.Knock) {
+        // We use the view flow for knock rooms as it allows us to reuse the request to join UX in RoomView
```

Comment:
I don't really understand this comment. In particular:
 * what's a view flow?
 * what's the request we are reusing?

I think I get it after about the 10th reading: is this accurate?

```suggestion
        // If the room is knockable, show the "View" button even if we are not a member; that
        // allows us to reuse the "request to join" UX in RoomView.
```

Reply:
Yes that is better English indeed